File: cpp/n383_RansomMagzine.cpp
// https://leetcode-cn.com/problems/ransom-note/submissions/
#include <iostream>
#include <vector>
#include <sstream>
#include <string>
#include <algorithm>
#include <unordered_map>
using namespace std;

class Solution{
public:
    bool canConstruct(string ransomNote, string magazine){
        for (size_t i = 0; i < ransomNote.size(); ++i){
            char ele = ransomNote[i];
            int pos = magazine.find(ele);
            if (pos == magazine.npos) return false;
            else magazine.erase(pos, 1);
        }
        return true;
    }
};

int main(){
    return 0;
}

File: python/n383_RansomMagzine.py
# https://leetcode-cn.com/problems/ransom-note/submissions/
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        l1 = list(ransomNote)
        l2 = list(magazine)
        for i in l1:
            if i not in l2:
                return False
            else:
                l2.remove(i)
        return True
        

if __name__ == "__main__":
    pass        